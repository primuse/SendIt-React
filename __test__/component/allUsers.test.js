import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import AllUsers from '../../src/components/allUsers.jsx';

const initialState = {
  auth: {
    isAuthenticated: true,
    user: {
      firstName: 'Tiku',
      lastName: 'Okoye',
      registered: '22-03-2019',
      email: 'cim@gmail.com'
    },
  },
  users: {
    allUsers: [
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      },
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      }
    ]
  },
  show: false
};

const mockStore = configureStore([thunk]);
const store = mockStore(initialState);


describe('All users component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <Provider store={store}>
          <AllUsers />
        </Provider>
      </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
