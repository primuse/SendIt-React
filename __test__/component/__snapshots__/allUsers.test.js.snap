// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All users component renders correctly 1`] = `
<main>
  <aside
    className=""
  >
    <div
      className="bg-blue"
      id="menu"
    >
      <div
        id="aside-profile"
      >
        <a
          href="/dashboard"
          onClick={[Function]}
        >
          <h3
            className="m-0 pb-10"
          >
            Send
            <span
              className="head-span"
            >
              IT
            </span>
          </h3>
        </a>
        <small
          className="d-block pb-10 white-3"
          id="user-id"
        >
          SD0
        </small>
        <small>
          <p
            className="m-0"
          >
            Welcome 
          </p>
          <p
            id="username"
          >
             
          </p>
        </small>
      </div>
      <ul
        className="mt-80"
      >
        <li>
          <img
            className="menu-ico"
            src="test-file-stub"
          />
          <a
            href="/dashboard"
            onClick={[Function]}
          >
            <span>
              Dashboard
            </span>
          </a>
        </li>
        <li>
          <img
            className="menu-ico"
            src="test-file-stub"
          />
          <a
            data-modal={true}
            data-target="#parcelmodal"
            href="/"
            onClick={[Function]}
          >
            <span>
              New Parcel
            </span>
          </a>
        </li>
        <li>
          <img
            className="menu-ico"
            src="test-file-stub"
          />
          <a
            href="/delivered_parcels"
            onClick={[Function]}
          >
            <span>
              Delivered Parcels
            </span>
          </a>
        </li>
        <li>
          <img
            className="menu-ico"
            src="test-file-stub"
          />
          <a
            href="/inTransit_parcels"
            onClick={[Function]}
          >
            <span>
              In-transit Parcels
            </span>
          </a>
        </li>
      </ul>
    </div>
  </aside>
  <section
    className="grey"
    id="dash"
  >
    <nav
      id="nav"
    >
      <a
        className="toggler mr-1"
        href="#"
        id="toggle"
        onClick={[Function]}
      >
        <img
          src="test-file-stub"
        />
      </a>
      <ul
        className="p-0"
      >
        <li>
          Dashboard
        </li>
      </ul>
      <ul
        className="p-0 ml-auto"
      >
        <li
          className="mr-1"
        >
          <a
            href="/profile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </li>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Logout
          </a>
        </li>
      </ul>
    </nav>
    <div
      id="main-content-page"
    >
      <div
        className="dash-cont"
      >
        <div
          id="table-info"
        >
          <a
            href="#"
          >
            All
          </a>
        </div>
        <div
          id="table-cont"
        >
          <table>
            <tbody>
              <tr
                id="table-head"
              >
                <th>
                  ID
                </th>
                <th>
                  First Name
                </th>
                <th>
                  Last Name
                </th>
                <th>
                  Admin
                </th>
                <th>
                  Email
                </th>
                <th>
                  Registered On
                </th>
              </tr>
              <tr>
                <td>
                  SD0
                  1
                </td>
                <td />
                <td />
                <td>
                  false
                </td>
                <td>
                  cim@gmail.com
                </td>
                <td>
                  22-03-2019
                </td>
                <td>
                  <a
                    className="btn xsm bg-bright-blue white"
                    href="/edit_user"
                    onClick={[Function]}
                  >
                    Edit
                  </a>
                </td>
              </tr>
              <tr>
                <td>
                  SD0
                  1
                </td>
                <td />
                <td />
                <td>
                  false
                </td>
                <td>
                  cim@gmail.com
                </td>
                <td>
                  22-03-2019
                </td>
                <td>
                  <a
                    className="btn xsm bg-bright-blue white"
                    href="/edit_user"
                    onClick={[Function]}
                  >
                    Edit
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        className="pagination"
      >
        <span>
          <a
            className="links"
            href="/"
            onClick={[Function]}
          >
             
            1
          </a>
        </span>
      </div>
    </div>
  </section>
</main>
`;
